icechunk/src/store.rs

```diff
@@ -935,6 +935,12 @@ async fn exists(key: &str, session: &Session) -> StoreResult<bool> {
         Key::Chunk { node_path, coords } => {
             match session.get_chunk_ref(&node_path, &coords).await {
                 Ok(r) => Ok(r.is_some()),
+                Err(SessionError {
+                    // this chunk belongs to a shard that has not been written
+                    // or is invalid
+                    kind: SessionErrorKind::InvalidIndexForSplitManifests { .. },
+                    ..
+                }) => Ok(false),
```

Comment:
should this really return false? I feel this is an error not a 'this key doesn't exist' response?

Reply:
This is some weird Zarr thing but I can't reproduce. commenting it out for now.

Reply:
Ah  this case (invalid coordinate) is now handled inside `get_old_chunk` . I refactored the code earlier, and this particular error handling isn't needed anymore. I will delete.